cfd_protocol/tests/cfds.rs

```diff
@@ -982,21 +928,82 @@ fn make_keypair(rng: &mut (impl RngCore + CryptoRng)) -> (SecretKey, PublicKey)
     (sk, pk)
 }
 
-/// Decompose a BIP340 signature into R and s.
-fn schnorrsig_decompose(signature: &schnorrsig::Signature) -> (schnorrsig::PublicKey, SecretKey) {
-    let bytes = signature.as_ref();
-
-    let nonce_pk = schnorrsig::PublicKey::from_slice(&bytes[0..32]).expect("R value in sig");
-    let s = SecretKey::from_slice(&bytes[32..64]).expect("s value in sig");
-
-    (nonce_pk, s)
+struct OliviaData {
+    pk: schnorrsig::PublicKey,
+    nonce_pks: Vec<schnorrsig::PublicKey>,
+    price: u64,
+    attestations: Vec<SecretKey>,
 }
 
-fn gen_price(rng: &mut impl RngCore, digits: &interval::Digits) -> u64 {
-    let (start, end) = digits.range().into_inner();
-    if start == end {
-        start
-    } else {
-        rng.gen_range(start, end)
+impl OliviaData {
+    /// An example of all the data necessary from `olivia` to test the
+    /// CFD protocol.
+    ///
+    /// Data comes from this event:
+    /// https://outcome.observer/h00.ooo/x/BitMEX/BXBT/2021-10-05T02:00:00.price[n:20].
+    pub fn example() -> Self {
```

Comment:
@thomaseizinger: I opted for this instead of using the code in `daemon/oracle.rs` because it's not trivial to use that code in several places:

- can't depend on the application from the library.
- moving it into the library is tricky because some of the types still belong in `daemon`, so its worth designing the sub-module/sub-crate properly.
- copying the `oracle_api` sub-module into the test folder is overkill and also doesn't work out of the box.

And more broadly, I think we should reinstate tests in which we have full control of the oracle, because then we can test each CET.